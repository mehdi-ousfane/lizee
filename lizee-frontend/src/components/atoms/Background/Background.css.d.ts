declare const styles: {
  readonly "background": string;
  readonly "filter": string;
};
export = styles;

