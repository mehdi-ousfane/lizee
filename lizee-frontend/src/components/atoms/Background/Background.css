.background {
  height: 100%;
  width: 100%;
  display: flex;
  position: relative;
  background-image: url('../../../assets/homeBackground.jpeg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  flex: 1;
  overflow: hidden;
}

.filter {
  flex: 1;
  display: flex;
}
