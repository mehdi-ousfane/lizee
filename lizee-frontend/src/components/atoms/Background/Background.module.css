.background {
  height: 100vh;
  width: 100%;
  display: flex;
  position: relative;
  background-image: url('../../../assets/background.jpeg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  flex: 1;
  overflow: hidden;
}

.filter {
  flex: 1;
  height: 100%;
  --bg-color: hsla(0, 0%, 8%, 0.4);
  --dot-color: hsla(0, 0%, 83%, 0.2);
  --dot-size: 0.5px;
  --dot-space: 10px;
  background-image: radial-gradient(
      var(--dot-color) var(--dot-size),
      transparent var(--dot-size)
    ),
    radial-gradient(
      var(--dot-color) var(--dot-size),
      transparent var(--dot-size)
    );
  background-color: var(--bg-color);
  background-position: 0 0, var(--dot-space) var(--dot-space);
  background-size: calc(var(--dot-space) * 2) calc(var(--dot-space) * 2);
}
