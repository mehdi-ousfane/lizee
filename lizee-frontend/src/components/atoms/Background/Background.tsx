import { ReactNode } from 'react';
import styles from './Background.css';

interface Props {
  children: ReactNode;
}

const Background = ({ children }: Props) => {
  return (
    <div className={styles.background}>
      <div className={styles.filter}>{children}</div>
    </div>
  );
};

export default Background;
