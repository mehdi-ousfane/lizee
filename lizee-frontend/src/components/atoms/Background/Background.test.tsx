import { render } from '@testing-library/react';
import Background from './Background';

describe('<Background />', () => {
  test('should render without crashing', () => {
    const { container } = render(<Background>Child content</Background>);
    expect(container).toMatchSnapshot();
  });

  test('should display children content', () => {
    const { getByText } = render(<Background>Child content</Background>);
    expect(getByText('Child content')).toBeTruthy();
  });
});
