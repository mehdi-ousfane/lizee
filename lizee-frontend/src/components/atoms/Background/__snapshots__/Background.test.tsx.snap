// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> should render without crashing 1`] = `
<div>
  <div
    class="background"
  >
    <div
      class="filter"
    >
      Child content
    </div>
  </div>
</div>
`;
